manager/integration/tests/test_rwx.py

```diff
@@ -1,12 +1,358 @@
+        delete_and_wait_pvc(core_api, pvc_manifest['metadata']['name'])
+
+    request.addfinalizer(finalizer)
+
+    return pvc_manifest
+
+
+@pytest.fixture(scope="module", autouse="True")
+def nfs_storage_class(request):
+    sc_manifest = {
+        'apiVersion': 'storage.k8s.io/v1',
+        'kind': 'StorageClass',
+        'metadata': {
+            'name': NFS_STORAGECLASS_NAME
+        },
+        'provisioner': 'nfs.longhorn.io',
+        'mountOptions': [
+            'vers=4.1',
+            'noresvport'
+        ]
+    }
+
+    def finalizer():
+        delete_storage_class(NFS_STORAGECLASS_NAME)
+
+    request.addfinalizer(finalizer)
+
+    return sc_manifest
+
+
+@pytest.fixture(scope="module", autouse="True")  # NOQA
+def core_api(request):
+    """
+    Create a new CoreV1API instance.
+    Returns:
+        A new CoreV1API Instance.
+    """
+    c = Configuration()
+    c.assert_hostname = False
+    Configuration.set_default(c)
+    k8sconfig.load_incluster_config()
+    core_api = k8sclient.CoreV1Api()
+
+    return core_api
+
+
+@pytest.fixture(scope="module", autouse="True")
+def apps_api(request):
+    """
+    Create a new AppsV1API instance.
+    Returns:
+        A new AppsV1API Instance.
+    """
+    c = Configuration()
+    c.assert_hostname = False
+    Configuration.set_default(c)
+    k8sconfig.load_incluster_config()
+    apps_api = k8sclient.AppsV1Api()
+
+    return apps_api
+
+
+@pytest.fixture(scope="module", autouse="True")
+def longhorn_nfs_provisioner_setup(core_api, apps_api, nfs_pvc,
+                                   nfs_storage_class, deployment_nfs):
+    """
+    Deploy longhorn-nfs-provisioner pod and volume
+    """
+
+    create_storage_class(nfs_storage_class)
+    core_api.create_namespaced_persistent_volume_claim(
+        body=nfs_pvc, namespace='longhorn-system')
+    apps_api.create_namespaced_deployment(namespace='longhorn-system',
+                                          body=deployment_nfs)
+    wait_deployment_replica_ready(apps_api, 'longhorn-nfs-provisioner', 1,
+                                  namespace='longhorn-system')
+
+
+def read_nfs_provisioner_data(api, command, pod_name, namespace):
+    """
+    Retrieve data from a Pod's volume.
+
+    Args:
+        api: An instance of CoreV1API.
+        pod_name: The name of the Pod.
+        command: The command to execute in the pod.
+        namespace: The namespace where the pod exists.
+
+    Returns:
+        The data contained within the volume.
+    """
+    exec_command = [
+        '/bin/sh',
+        '-c',
+        command
+    ]
+    with timeout(seconds=STREAM_EXEC_TIMEOUT,
+                 error_message='Timeout on executing stream read'):
+        return stream(
+            api.connect_get_namespaced_pod_exec, pod_name, namespace,
+            command=exec_command, stderr=True, stdin=False, stdout=True,
+            tty=False)
+
+
+@pytest.mark.skip(reason="Waiting for rwx enhancement")  # NOQA
```

Comment:
Do we still need to skip the test?

Reply:
Created the PR https://github.com/longhorn/longhorn-manager/pull/671, I think once that is merged we can remove the skip.